homeassistant/components/sonos/media_player.py

```diff
@@ -725,8 +725,13 @@ def update(self, now=None):
 
     def update_media(self, event=None):
         """Update information about currently playing media."""
-        transport_info = self.soco.get_current_transport_info()
-        new_status = transport_info.get("current_transport_state")
+        variables = event and event.variables
+
+        if variables:
+            new_status = variables["transport_state"]
```

Comment:
First review in this repo, so please forgive me if my comments are irrelevant

1. Isn't this better if there is also a check if `transport_state` is present in variables, so it becomes `if variables and "transport_state" in variables:`
2. Shouldn't there be a unit test which has to be changed/extended for this?

Reply:
Welcome :-).

1. It is assumed to always be present so I think it's fine to fail fast if that assumption is not met.
2. It is not a requirement to add tests for integrations and this part was never tested.